### PR TITLE
Remove bin_io from Public_key and Public_key.Compressed

### PR DESCRIPTION
**Pull request by psteckler**
Removed `bin_io` for `Public_key` and its `Compressed` submodule, propagate that change.

Lots of types are marked with TODO to version them.

There are some unfortunate `include`s in modules to get `bin_io`, needed because of existing monkeypatching.

Checklist:

- [ ] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [X] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: src/app/cli/src/coda_main.ml [hunk 1]
```diff
@@ -51,15 +51,7 @@ end
...
-  let of_digest = Ledger_hash.of_digest
-
-  let merge = Ledger_hash.merge
-
-  let to_bytes = Ledger_hash.to_bytes
-
-  let of_digest = Ledger_hash.of_digest
-
-  let merge = Ledger_hash.merge
+  let of_digest, merge, to_bytes = Ledger_hash.(of_digest, merge, to_bytes)
```
Comment (cmr) [inline src/app/cli/src/coda_main.ml]: oh, very nice shorthand. I didn't realize module items take patterns on the left hand side. of course they do though.

Comment (bkase) [reply src/app/cli/src/coda_main.ml]: wow that's really cool

Comment (bkase) [reply src/app/cli/src/coda_main.ml]: this changes everything

Comment (psteckler) [reply src/app/cli/src/coda_main.ml]: I'm glad to be of service. :-)

**Pull request merged**